src/main/java/adris/altoclef/tasks/movement/GetToXZWithElytraTask.java

```diff
@@ -0,0 +1,251 @@
+package adris.altoclef.tasks.movement;
+
+import adris.altoclef.AltoClef;
+import adris.altoclef.Debug;
+import adris.altoclef.TaskCatalogue;
+import adris.altoclef.tasks.slot.MoveItemToSlotFromInventoryTask;
+import adris.altoclef.tasks.misc.RepairToolTask;
+import adris.altoclef.tasks.slot.ClickSlotTask;
+import adris.altoclef.tasks.slot.EnsureFreeInventorySlotTask;
+import adris.altoclef.tasksystem.Task;
+import adris.altoclef.util.ItemTarget;
+import adris.altoclef.util.csharpisbetter.TimerGame;
+import adris.altoclef.util.helpers.LookHelper;
+import adris.altoclef.util.helpers.StorageHelper;
+import adris.altoclef.util.helpers.WorldHelper;
+import adris.altoclef.util.slots.PlayerSlot;
+import adris.altoclef.util.slots.Slot;
+import baritone.api.utils.Rotation;
+import baritone.api.utils.input.Input;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Vec3d;
+import java.util.List;
+
+public class GetToXZWithElytraTask extends Task {
+
+    private final int _x, _z;
+    private boolean _isFinished;
+    private boolean _isMovingElytra = false;
+    private boolean _isCollectingFireWork = false;
+    private boolean _isFlyRunning = false;
+    private boolean _repairElytra = false;
+    private double _oldCoordsY;
+    private int _yGoal = 0;
+    private int _fx = 0;
+    private int _fz = 0;
+    private final TimerGame _fireWorkTimer = new TimerGame(3);
+    private final TimerGame _messageProgessTimer = new TimerGame(3);
+    private final TimerGame _getToSurfaceTimer = new TimerGame(2);
+    private final TimerGame _jumpTimer = new TimerGame(0.1);
+
+    public GetToXZWithElytraTask(int x, int z) {
+        _x = x;
+        _z = z;
+    }
+    @Override
+    protected void onStart(AltoClef mod) {
+        _jumpTimer.reset();
+        _fireWorkTimer.reset();
+        _messageProgessTimer.reset();
+    }
+
+    @Override
+    protected Task onTick(AltoClef mod) {
+        double dist = mod.getPlayer().getPos().distanceTo(new Vec3d(_x, mod.getPlayer().getPos().y, _z)); //Calculate distance
+        if (!_isFlyRunning) { //If we are already flying, jump that code section
+            _fx = 0;
+            _fz = 0;
+            mod.getBehaviour().disableDefence(false); //Enable mob defence
+            if ((int)dist == 0) { //We are where we need to go !
+                _isFinished = true; //End the task
```

Comment:
Why is mob defense even disabled? Doesnt that mean phantoms and mobs that shoot can get us mid-flight?

Reply:
It disable mob defense while flying because the bot try to avoid it's own firework when using it. and i think the player move faster than the phantoms while flying